### PR TITLE
Clean up old Conda tutorial and update Tykky

### PR DESCRIPTION
**Pull request by rkronberg**
https://csc-guide-preview.rahtiapp.fi/origin/clean-conda-tutorial/support/tutorials/conda/
https://csc-guide-preview.rahtiapp.fi/origin/clean-conda-tutorial/computing/containers/tykky/

### CODE REVIEW

File: docs/support/tutorials/conda.md [hunk 1]
```diff
@@ -1,424 +1,52 @@
...
+    Check if any of [CSC's pre-installed environments](../apps/index.md) would
+    be suitable for your project. If the existing environment is missing a few
+    critical packages, you can often install the missing packages on your own.
+
+    Our [Python](../apps/python.md#installing-python-packages-to-existing-modules)
+    and [R](../apps/r-env.md#r-package-installations) pages also contain further
+    details on how to install your own packages to our modules. You can also
+    [contact CSC Service Desk](contact.md) with requests for missing packages.
+
+2. **Wrap your Conda or pip environment using CSC's Tykky tool**
```
Comment (ktiits) [inline docs/support/tutorials/conda.md:33]:
```suggestion
2. **Create containerized Conda or pip environment using CSC's Tykky tool**
```

File: docs/support/tutorials/conda.md [hunk 2]
```diff
@@ -1,424 +1,52 @@
...
+
+    CSC has developed [a tool for wrapping existing Conda or pip
```
Comment (ktiits) [inline docs/support/tutorials/conda.md:35]:
```suggestion
    CSC has developed [a tool for wrapping Conda or pip
```

File: docs/support/tutorials/conda.md [hunk 3]
```diff
@@ -1,424 +1,52 @@
...
+    This is a great alternative for developing software locally on a workstation,
+    and then deploying it on another workstation, cluster, or on cloud platforms.
+    CSC's supercomputers support Apptainer containers, which are are just single
+    large files for Lustre, thus avoiding much of the problems. Many software
+    projects offer Docker containers which can often be easily converted into
+    Apptainer format. Inside the container you can naturally use for example
+    Conda to manage the packages without causing any file system issues.
+
+    Read our documentation on [how to create your own Apptainer
+    container](../computing/containers/creating.md)
```
Comment (ktiits) [inline docs/support/tutorials/conda.md]:
```suggestion
    Use [Tykky](../computing/containers/tykky.md#container-based-installations) to convert an existing Docker container to Apptainer container or read our documentation on [how to create your own Apptainer
    container](../computing/containers/creating.md)
```

File: docs/support/tutorials/conda.md [hunk 4]
```diff
@@ -1,424 +1,52 @@
...
-[environment.yaml]: https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#creating-an-environment-from-an-environment-yml-file
-[Spack]: https://spack.io
-[Miniconda]: https://docs.conda.io/en/latest/miniconda.html
-[Anaconda]: https://www.anaconda.com/distribution
-[conda-docs-env.yaml]: conda/conda-docs-env-1.0.yaml
-[mkdocs]: https://www.mkdocs.org
-[c-ide.yaml]: conda/c-ide.yaml
-[GNU Global]: https://www.gnu.org/software/global
-[bash_profile_extras.sh]: conda/bash_profile_extras.sh
+!!! warning "Do not install Conda environments directly on the shared file system!"
```
Comment (ktiits) [inline docs/support/tutorials/conda.md:3]:
```suggestion
!!! warning "Do not install Conda environments directly to supercomputers!"
```

File: docs/support/tutorials/conda.md [hunk 5]
```diff
@@ -1,424 +1,52 @@
 # Conda best practices
```
Comment (ktiits) [inline docs/support/tutorials/conda.md:1]:
```suggestion
# Conda best practices for supercomputers
```

File: docs/support/tutorials/conda.md [hunk 6]
```diff
@@ -1,424 +1,52 @@
...
-[bash_profile_extras.sh]: conda/bash_profile_extras.sh
+!!! warning "Do not install Conda environments directly on the shared file system!"
+    [CSC has deprecated the use of Conda environments](../../computing/usage-policy.md)
+    that are installed directly on the shared file system (e.g. `/scratch`, `/projappl`,
+    `$HOME`) on CSC supercomputers. This is due to performance issues of Conda-based
+    environments on parallel file systems, causing long start-up delays and system-wide
+    slowdowns when using Python scripts. Users should either containerize their
+    Conda-based installations or consider alternative means of installation. See
+    also the [Tykky container wrapper](../../computing/containers/tykky.md), a tool
+    using which you can easily containerize your Conda installations.
```
Comment (ktiits) [inline docs/support/tutorials/conda.md]:
```suggestion
    slowdowns when using Python scripts. 
```

File: docs/computing/containers/tykky.md [hunk 7]
```diff
@@ -130,74 +141,85 @@ Type "help", "copyright", "credits" or "license" for more information.
...
+  Apptainer/Singularity](https://docs.sylabs.io/guides/3.7/user-guide/cli/singularity_pull.html)
+  (e.g `docker://` `oras://`)
+- `-w` needs to be an absolute path (or comma-separated list) inside the container.
+  Wrappers will then be automatically created for the executables in the target
+  directories / for the target path. If you do not know the path of the executables
+  in the container, open a shell inside the container and use the [which
+  command](https://linuxize.com/post/linux-which-command/). To open a shell:
+  	- In case of existing local Apptainer/Singularity file: `singularity shell image.sif`.
+  	- In case of Docker or non-local Apptainer/Singularity file, create first the
+      installation with some path and then start with created `_debug_shell`.
```
Comment (ktiits) [inline docs/computing/containers/tykky.md:217]: ## Memory errors

With very big installations the resources available on the login-node might not be enough and Tykky might fail with a `MemoryError`. In this case the installation needs to be done on a compute node in [interactive session](../running/interactive-usage.md#sinteractive-in-puhti).

```
# Start interactive session, here with 12 Gb memory and 15 Gb local disc, increase if needed.
sinteractive -m 12000 -d 15

# Define where to store temporary files of Apptainer
export APPTAINER_CACHEDIR=$LOCAL_SCRATCH
export APPTAINER_TMPDIR=$LOCAL_SCRATCH

cd <your_installation_folder>
# Run the Tykky commands as described above
```

Comment (ktiits) [reply docs/computing/containers/tykky.md:217]: @Nortamo Is this correct? Copied from Ari-Matti's RT ticket and I have used this sometimes, but I think myself did not use these export rows.. But not sure..

Comment (rkronberg) [reply docs/computing/containers/tykky.md:217]: Tykky appears to know to use `$LOCAL_SCRATCH` by default (if nvme is requested), so these exports shouldn't be necessary.

**Review by ktiits — APPROVED** [review]

**Pull request merged**